components/net/connector.rs

```diff
@@ -50,9 +50,31 @@ impl NetworkConnector for HttpsConnector {
 
 pub type Connector = HttpsConnector;
 
-pub fn create_ssl_client(ca_file: &PathBuf) -> OpensslClient {
+pub fn create_ssl_client(certs: &str) -> OpensslClient {
+    let mut certs = certs.to_string();
+    // certs include multiple certificates. We could add all of them at once,
+    // but if any of them were already added, openssl would fail to insert all
+    // of them.
     let mut ssl_connector_builder = SslConnectorBuilder::new(SslMethod::tls()).unwrap();
-    ssl_connector_builder.set_ca_file(ca_file).expect("could not set CA file");
+    loop {
+        if let Some(index) = certs.rfind("-----BEGIN CERTIFICATE-----") {
+            let cert = certs.split_off(index);
+            let cert = x509::X509::from_pem(cert.as_bytes()).unwrap();
+            ssl_connector_builder.cert_store_mut().add_cert(cert).or_else(|e| {
+                let v: Option<Option<&str>> = e.errors().iter().nth(0).map(|e| e.reason());
+                if v == Some(Some("cert already in hash table")) {
+                    warn!("Cert already in hash table. Ignoring.");
+                    // Ignore error X509_R_CERT_ALREADY_IN_HASH_TABLE which means the
+                    // certificate is already in the store.
+                    Ok(())
+                } else {
+                    Err(e)
+                }
+            }).expect("could not set CA file");
+        } else {
+            break;
```

Comment:
It looks like this might skip the last certificate in the file, because it's not followed by a `BEGIN CERTIFICATE` line.

Reply:
I use `END CERTIFICATE` now.